### PR TITLE
Toggle theme

### PR DESCRIPTION
**Pull request by boonepeter**
# Description

I am finally getting around to issue #669. This does not fix that issue, but it enables the user to toggle between light and dark modes from the GUI View menu. 

 I tried to make sure that the check mark would get updated if the palette was updated elsewhere. That seemed a little hack-y to me so I welcome any other ideas if there is a better way. Still learning the Qt world.

I will borrow a mac and try to fix the weird updating bug that others are seeing.

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by tlambert03** [comment]
Hi @boonepeter.  thanks for the PR!  this looks/works good for me.  
Ultimately, I think I'd like to see this functionality in a preferences menu, with a dropdown selection instead of a boolean choice (in case we expand theme options, or allow plugins).  I implemented that in the settings PR #834 ([here is an example](https://github.com/napari/napari/pull/834#issuecomment-569544298))... but I think it might be a while until there is sufficient demand to get that in.  So I have no problem with this solution for now.

will leave it for @sofroniewn to also review.  And you'll need to fix the black/flake8 errors before it can merge.

thanks again!

**Comment by boonepeter** [comment]
Thanks @tlambert03! Your #834 looks pretty slick and is definitely the way to go. I definitely won't be offended if this doesn't make the cut as it is just a small thing I wanted to add while looking into that other issue.

**Comment by tlambert03** [comment]
yeah, but that could take a while, and this is minimal.  The only possible argument against it is that we might eventually remove it.  but I don't mind that.  I'll approve this but leave the final decision to @sofroniewn.

**Review by tlambert03 — APPROVED** [review]

File: napari/_qt/qt_main_window.py
Comment (sofroniewn) [inline napari/_qt/qt_main_window.py]: I'd prefer to see this function look like the toggle_theme function in our `napari/keybindings.py` https://github.com/napari/napari/blob/c01024f151546f7a2ada9727572efb3e65fa4f0f/napari/keybindings.py#L15
Can you replace the code here with that code (keeping the name `_toggle_theme` and remove the code around `light` and `dark`. The reason for this is that if people add more themes it will allow them to cycle through them instead of just having to alternate between our two fixed ones

File: napari/_qt/qt_main_window.py
Comment (sofroniewn) [inline napari/_qt/qt_main_window.py]: I think I'd rather see the `QAction` be called `Toggle theme`, with the status tip `Toggle theme` and the keybinding `Control+Shift+T` as is used https://github.com/napari/napari/blob/c01024f151546f7a2ada9727572efb3e65fa4f0f/napari/keybindings.py#L15. There shouldn't be any special references to light / dark here

Once this is done you can remove the code here https://github.com/napari/napari/blob/c01024f151546f7a2ada9727572efb3e65fa4f0f/napari/keybindings.py#L15

File: napari/_qt/qt_main_window.py
Comment (sofroniewn) [reply napari/_qt/qt_main_window.py]: Ah, also this function should maybe live inside the `ViewerModel` https://github.com/napari/napari/blob/master/napari/components/viewer_model.py instead of `qt_main_window` as it only needs to reference the Viewer

**Review by sofroniewn — CHANGES_REQUESTED** [review]
Hi @boonepeter thanks for adding this! I've requested some changes as we have some functionality for toggling themes right now that allows you to cycle through our current themes here https://github.com/napari/napari/blob/c01024f151546f7a2ada9727572efb3e65fa4f0f/napari/keybindings.py#L15, and I think we can move that code inside the ViewerModel and then reuse it in a drop down menu.

The code allows for the cycling through themes - which maybe a little annoying if you have many themes added, but is also a little more flexible than if you just have the two default ones. I think in the long run we'll probably have a solution more along the lines of #834.

@tlambert03 does this approach seem reasonable too?

File: napari/_qt/qt_main_window.py
Comment (boonepeter) [reply napari/_qt/qt_main_window.py]: I like it. I copied this and moved it to `viewer_model`

File: napari/_qt/qt_main_window.py
Comment (boonepeter) [reply napari/_qt/qt_main_window.py]: I renamed this and removed the keybinding. The keybinding no longer shows up in the "keybindings help" window, but I guess that isn't an issue since it is visible in the main menu. 

I changed `Ctrl-Shift-O` to `Ctrl+Shift+O` for the open folder shortcut so that Qt recognizes this and displays the shortcut it in the menu.

**Review by sofroniewn — APPROVED** [review]
Thanks for making those changes so quickly @boonepeter. This looks great, I have confirmed it works for me. I will approve and then merge shortly.

**Pull request merged**
